Lib/fontTools/merge.py

```diff
@@ -975,20 +1089,52 @@ def merge(self, fontfiles):
 
 		return mega
 
-	def _mergeGlyphOrders(self, glyphOrders):
+	def _mergeGlyphOrders(self, glyphOrders, cffTables):
 		"""Modifies passed-in glyphOrders to reflect new glyph names.
 		Returns glyphOrder for the merged font."""
 		# Simply append font index to the glyph name for now.
 		# TODO Even this simplistic numbering can result in conflicts.
 		# But then again, we have to improve this soon anyway.
+
+		# If we have CFF tables,
+		# rename topDictIndex charStrings
+		for n,cffTable in enumerate(cffTables):
+			td = cffTable.cff.topDictIndex[0]
+			d = {}
+			for i, (glyphName, v) in enumerate(td.CharStrings.charStrings.items()):
+				if glyphName not in ['.notdef']:
+					glyphName += "#" + repr(n)
+				elif n > 0:
+					glyphName += "#" + repr(n)
+				d[glyphName] = v
+			cffTable.cff.topDictIndex[0].CharStrings.charStrings = d
+
 		mega = []
-		for n,glyphOrder in enumerate(glyphOrders):
-			for i,glyphName in enumerate(glyphOrder):
-				glyphName += "#" + repr(n)
-				glyphOrder[i] = glyphName
-				mega.append(glyphName)
+
+		# If we have CFF tables,
+		# make sure to keep .notdef as first glyph,
+		# don't rename it, and skip any additional .notdef
```

Comment:
the special handling of .notdef could be extended to non-CFF as well.

Reply:
ok